### DISCUSSION TITLE
Mountain Duck modification time not updated when using MacVim

### DISCUSSION THREAD

**Discussion by jknockaert** [iterate-ch/cyberduck#15467] [Q&A]
**Describe the bug**
When opening a file that was synced by Mountain Duck using MacVim and making modifications, the modification timestamp does not get updated after saving (`+w` or `+x` in MacVim).

**To Reproduce**
1. Open a connection to a server (using sftp in online connection mode and with index files and buffering disabled)
2. Open a file in MacVim
3. Edit the file and save
4. Modification timestamp not updated nor in the local copy nor on the server

**Expected behavior**
Modification timestamp updated both locally and remotely.

**Desktop (please complete the following information):**
 - OS: macOS 14.1.2
 - MacVim: r178 (Vim 9.0.1897)
 - Version: Mountain Duck 4.15.1 (21679)

**Comment by rwaldspur** [comment]
This behavior is also present when editing a PDF in macOS application "PDF Expert", with server side as Google Drive or DropBox.

Substitute "PDF Expert" for "MacVim" in above description.

Also, if you do a "touch" command in the Terminal window, it will not update the modification date.

OS: macOS v13.6.1
PDFExpert v3.7.1
MountainDuck v4.15.2 (21684)
CPU: Apple M2

**Comment by dkocher** [comment]
> This behavior is also present when editing a PDF in macOS application "PDF Expert", with server side as Google Drive or DropBox.

@rwaldspur Are you working in _Online_ connect mode as well?

**Comment by rwaldspur** [comment]
No, thanks for that clarification.  I am in OFFLINE mode, though I believe I tested online mode as well.

Here are my sync settings.
<img width="578" alt="Mountain Duck Sync settings" src="https://github.com/iterate-ch/cyberduck/assets/29511202/dfe739a1-b3f0-4d49-8bf0-c8e7a0e79cd2">

**Reply by rwaldspur** [reply]
I did in fact test online mode as well, and it also doesn't change the modified date.

Worth noting -- When I duplicate a File in the macOS finder in online mode, the creation date is changed to the current time of duplication, instead of the original creation date.  This is different behavior than the offline mode.

I realize that preservation of Creation Dates was a recent change, and I'm quite appreciate of that.  Not sure what the expected behavior should be in online mode for it.

**Comment by dkocher** [comment]
@jknockaert `vim` does not explicitly set the modification date when writing the file. Actually it moves away the previous versions and then truncates and writes to a new file with the same name. I cannot seem to reproduce the modification date not updating but instead the modification date is set to the current time determined by the SSH (SFTP) server.

**Comment by sir-oliver** [comment]
I experience the same behaviour with VS Code, but it happens only sporadically and is not really reproducable.
Sometimes saving a file won't update the modification date. As the web-application only refreshes when the timestamp has changed, it is a bit frustrating to always double check with a SSH session if the timestamp was updated correctly.

OS: macOS v14.4
VSCode v1.75.0
MountainDuck v4.15.4 (21882)
